node/bft/events/src/helpers/codec.rs

```diff
@@ -83,11 +84,15 @@ impl<N: Network> Decoder for EventCodec<N> {
             Some(bytes) => bytes,
             None => return Ok(None),
         };
-
+        let bytes_len = bytes.len() as f64;
         // Convert the bytes to an event, or fail if it is not valid.
         let reader = bytes.reader();
         match Event::read_le(reader) {
-            Ok(event) => Ok(Some(event)),
+            Ok(event) => {
+                #[cfg(feature = "metrics")]
+                histogram!(metrics::tcp::TCP_GATEWAY, "event" => event.name()).record(bytes_len);
```

Comment:
Use the histogram from `snarkvm_metrics`

Reply:
In snarkvm_metrics, you can only update a histogram with a static name and value, which would require creating a separate metric name for every event type-- using just the regular histogram macro allows you to add labels which keeps the number of separate metrics lower and allows for easier analysis in Prometheus
I tried to call the original metrics crate histogram macro from here but it looked like I would need to add it to expose it in the snarkvm_metrics crate 😅

Reply:
Got it, thanks for the clarification! I'll take a look at this when things are more stable.